### PR TITLE
Culling

### PR DESCRIPTION
**Pull request by Przemog1**

### CODE REVIEW

File: include/nbl/asset/utils/IMeshPacker.h [hunk 1]
```diff
@@ -239,7 +239,7 @@ class IMeshPacker : public IMeshPackerBase
     //TODO: functions: constructTriangleBatches, convertIdxBufferToTriangles, deinterleaveAndCopyAttribute and deinterleaveAndCopyPerInstanceAttribute
     //will not work with IGPUMeshBuffer as MeshBufferType, move it to new `ICPUMeshPacker`
 
-    TriangleBatches constructTriangleBatches(const MeshBufferType* meshBuffer, IdxBufferParams idxBufferParams, core::aabbox3df* aabbs) const
+    TriangleBatches constructTriangleBatches(const MeshBufferType* meshBuffer, IdxBufferParams idxBufferParams, core::aabbox3df*& aabbs) const
```
Comment (devshgraphicsprogramming) [inline include/nbl/asset/utils/IMeshPacker.h:242]: why a reference to a pointer?

Comment (Przemog1) [reply include/nbl/asset/utils/IMeshPacker.h:242]: otherwise it doesn't work

Comment (devshgraphicsprogramming) [reply include/nbl/asset/utils/IMeshPacker.h:242]: you're changing the pointer?

Comment (Przemog1) [reply include/nbl/asset/utils/IMeshPacker.h:242]: copy of pointer passed to the `commit` function, yes

Comment (Przemog1) [reply include/nbl/asset/utils/IMeshPacker.h:242]: original pointer remains unchanged

File: examples_tests/41.VisibilityBuffer/main.cpp [hunk 2]
```diff
@@ -526,6 +582,28 @@ int main()
...
+                        batchCullDataEnd->aabbMaxEdge.z = aabbs[aabbIdx].MaxEdge.Z;
+
+                        batchCullDataEnd->drawCommandGUID = packedMeshBufferData.mdiParameterOffset + i;
+
+                        draw3DLine->enqueueBox(dbgLines, aabbs[aabbIdx], 0.0f, 0.0f, 0.0f, 1.0f, core::matrix3x4SIMD());
+
+                        batchCullDataEnd++;
+                        aabbIdx++;
+                    }
+                }
```
Comment (devshgraphicsprogramming) [inline examples_tests/41.VisibilityBuffer/main.cpp:605]: can you hide this under a DEBUG ifdef?

File: include/nbl/asset/utils/IMeshPacker.h (hunk 1 above)
Comment (devshgraphicsprogramming) [reply include/nbl/asset/utils/IMeshPacker.h:242]: ok I see the usage and whats going on

**Pull request merged**